src/Adapter/Module/Tab/ModuleTabRegister.php

```diff
@@ -0,0 +1,237 @@
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\Translation\TranslatorInterface;
+use TabCore as Tab;
+
+class ModuleTabRegister
+{
+    private $defaultParent = 'DEFAULT';
+    /**
+     * @var LangRepository
+     */
+    protected $langRepository;
+
+    /**
+     * @var TabRepository
+     */
+    protected $tabRepository;
+
+    /**
+     * @var Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var Symfony\Component\Translation\TranslatorInterface
+     */
+    private $translator;
+    
+    /**
+     * @var Finder
+     */
+    private $finder;
+    
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+    
+    public function __construct(TabRepository $tabRepository, LangRepository $langRepository, LoggerInterface $logger, TranslatorInterface $translator, Finder $finder, Filesystem $filesystem)
+    {
+        $this->langRepository = $langRepository;
+        $this->tabRepository = $tabRepository;
+        $this->logger = $logger;
+        $this->translator = $translator;
+        $this->finder = $finder;
+        $this->filesystem = $filesystem;
+    }
+    
+    /**
+     * Fetch module-defined tabs and find undeclared ModuleAdminControllers.
+     *
+     * This is done automatically as part of the module installation.
+     * @param Module $module
+     * 
+     */
+    public function registerTabs(Module $module)
+    {
+        $tabs = $this->addUndeclaredTabs($module->get('name'), $module->getInstance()->getTabs());
+        
+        foreach ($tabs as $tab) {
+            try {
+                $this->registerTab($module, new ParameterBag($tab));
+            } catch (Exception $e) {
+                $this->logger->error($e->getMessage());
+            }
+        }
+
+    }
+    
+    /**
+     * Looks for ModuleAdminControllers not declared as Tab and 
+     * add them to the list to register
+     * 
+     * @param string $moduleName
+     * @param array $tabs
+     * @return array
+     */
+    protected function addUndeclaredTabs($moduleName, array $tabs)
+    {
+        // Function to get only class name from tabs already declared
+        $tabsNames = array_map(function($tab) {
+            if (array_key_exists('class_name', $tab)) {
+                return $tab['class_name'];
+            }
+        }, $tabs);
+        
+        foreach ($this->getModuleAdminControllersFilename($moduleName) as $adminControllerFileName) {
+            $adminControllerName = str_replace('Controller.php', '', $adminControllerFileName);
+            if (in_array($adminControllerName, $tabsNames)) {
+                continue;
+            }
+            
+            $tabs[] = array(
+                'class_name' => $adminControllerName,
```

Comment:
To be added:
```php
'visible' => false,
``` 

ModuleAdminController are not going to be visible. We just want them to be callable for display in the BackOffice